### PR TITLE
MULE-14866: Only asynchronous notifications should receive a copy of the MuleEvent.

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

File: core/src/test/java/org/mule/construct/processor/PipelineMessageNotificationTestCase.java [hunk 1]
```diff
@@ -383,7 +383,7 @@ public MuleEvent process(MuleEvent event) throws MuleException
             public boolean test()
             {
                 verify(notificationManager, times(1)).fireNotification(
-                        argThat(new PipelineMessageNotificiationArgumentMatcher(PROCESS_START, false, event)));
+                        argThat(new PipelineMessageNotificiationArgumentMatcher(PROCESS_START, false, null)));
```
Comment (elrodro83) [inline core/src/test/java/org/mule/construct/processor/PipelineMessageNotificationTestCase.java]: why?

File: core/src/test/java/org/mule/mule/enricher/MessageEnricherTestCase.java [hunk 2]
```diff
@@ -183,6 +186,44 @@ public MuleEvent process(MuleEvent event) throws MuleException
...
+                        }
+                    }).start();
+                }
+            }
+        });
+        MessageEnricher enricher = createEnricher(new SensingNullMessageProcessor());
+        final MuleEvent in  = createBlockingEvent();
+        try
+        {
+            processEnricherInChain(enricher, in);
```
Comment (elrodro83) [inline core/src/test/java/org/mule/mule/enricher/MessageEnricherTestCase.java]: don't need the try-catch here

File: core/src/main/java/org/mule/api/context/notification/ServerNotification.java [hunk 3]
```diff
@@ -264,4 +266,18 @@ private static Object verifyNullResource(Object resource)
...
     }
 
+    public static boolean resolveEventReference(MuleEvent event)
+    {
+        if (event instanceof VoidMuleEvent)
+        {
+            return true;
+        }
+        
+        if (event instanceof DefaultMuleEvent)
```
Comment (elrodro83) [inline core/src/main/java/org/mule/api/context/notification/ServerNotification.java]: only have this check and return true otherwise

**Comment by fsgonz** [comment]
I have to reconsider the general solution.

**Pull request closed without merging**